test/integration/rest-basic/index.spec.ts

```diff
@@ -43,11 +44,12 @@ test('rest runtime basics', async ({ page, localApp }) => {
   await expect(page.getByText('query4 authorization: foo')).toBeVisible();
 
   const envFilePath = path.resolve(localApp.dir, './.env');
-  await fs.writeFile(envFilePath, 'TEST_VAR=bar');
-
-  await page.reload();
-
-  await expect(page.getByText('query4 authorization: bar')).toBeVisible();
+  await withTemporaryEdits(envFilePath, async () => {
```

Comment:
Seems good! A possible alternative could maybe be a utility function that we could pass the file path and content to, which would write to the file, and would return a function that we could call in order to undo that edit. Not sure if that would be cleaner.

Reply:
This is sort of the poor person's version of [explicit resource management](https://github.com/tc39/proposal-explicit-resource-management)

Reply:
Oh I didn't consider the test failing, this seems solid then!